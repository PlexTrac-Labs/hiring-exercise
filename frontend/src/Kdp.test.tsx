import React from "react";
import renderer from "react-test-renderer";
import Link from "./Kdp";

import axios from "axios";
import {
  render,
  cleanup,
  act,
  findByTestId,
  waitFor,
} from "@testing-library/react";

jest.mock("axios");
const mockedAxios = axios as jest.Mocked<typeof axios>;

afterEach(cleanup);

it("renders", async () => {
  mockedAxios.get.mockResolvedValue({
    data: [
      { id: 1, title: "title 1" },
      { id: 2, title: "title 2" },
    ],
  });
  const { getByTestId, asFragment } = render(<Link />);

  const listNode = await waitFor(() => getByTestId("list"));
  expect(listNode.children).toHaveLength(2);
  expect(asFragment()).toMatchSnapshot();
});
