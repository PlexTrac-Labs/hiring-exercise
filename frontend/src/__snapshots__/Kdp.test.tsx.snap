// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
          
    <ul
      data-testid="list"
    >
              
      <li>
        title 1
      </li>
      <li>
        title 2
      </li>
            
    </ul>
        
  </div>
</DocumentFragment>
`;
